src/App.php

```diff
@@ -516,12 +515,12 @@ public function run()
             if (isset($_GET['__atk_callback']) && $this->catch_runaway_callbacks) {
                 throw new Exception('Callback requested, but never reached. You may be missing some arguments in request URL.');
             }
-            echo $this->html->template->render();
+
+            $output = $this->html->template->render();
```

Comment:
Probably good to initialize `$output` for cases this line is not reached

Reply:
You are right! But still working on other unclosed buffers, see warnings in unit tests.